src/GA4ServiceProvider.php

```diff
@@ -21,14 +22,31 @@ public function configurePackage(Package $package): void
             ->hasRoute('web');
     }
 
-    public function registeringPackage()
+    public function registeringPackage(): void
     {
         $this->app->bind('ga4', function () {
-            return new GA4MeasurementProtocol();
+            if (config('google-analytics-4-measurement-protocol.measurement_id') === null
+                || config('google-analytics-4-measurement-protocol.api_secret') === null
+            ) {
+                throw new Exception('Please set .env variables for Google Analytics 4 Measurement Protocol as per the readme file first.');
+            }
+
+            return new GA4MeasurementProtocol(
+                config('google-analytics-4-measurement-protocol.measurement_id'),
+                config('google-analytics-4-measurement-protocol.api_secret'),
+                function () {
+                    $clientId = session(config('google-analytics-4-measurement-protocol.client_id_session_key'));
+                    if (empty($clientId)) {
+                        throw new Exception('Please use the package provided blade directive or set client_id manually before posting an event.');
+                    }
+
+                    return $clientId;
+                }
```

Comment:
What is the benefit of passing the closure from here instead of keeping the login inside the facade?

Reply:
Service class should not  know about configuration. But provider must configuring service.